frontend/catalyst/compilation_pipelines.py

```diff
@@ -531,7 +531,7 @@ def _maybe_promote(self, function, *args):
         if needs_compile:
             if self.user_typed:
                 msg = "Provided arguments did not match declared signature, recompiling..."
-                warnings.warn(msg, UserWarning)
+                warnings.warn(msg, ResourceWarning)
```

Comment:
Just curious why `ResourceWarning`, is there a leaked resource?

Reply:
I can change it back. Originally I changed it to avoid some failing tests which are now resolved with #146 . But I think `ResourceWarning` can also be used to indicate that it is taking more computational resources than expected (I.e., we are recompiling therefore wasting resources). Not a big deal to change back.